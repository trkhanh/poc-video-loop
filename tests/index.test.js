import React from 'react'
import { render } from '@testing-library/react'
import VideoLooper from '../src'

describe('VideoLooper', () => {

    test('exists', () => {
        expect(VideoLooper).toBeDefined()
    })

    test('matches Snapshot', async () => {
        const { container } = render(<VideoLooper source='sample.mp4' start={4.31} end={9.48} muted={false} />)
        expect(container).toMatchSnapshot()
    })

    test('contain video element', async () => {
        const { container } = render(<VideoLooper source='sample.mp4' start={4.31} end={9.48} muted={false} />)
        expect(container.querySelector('video')).toBeInTheDocument()
    })

    test('debug view contains header text', async () => {
        const { getAllByText } = render(<VideoLooper source='sample.mp4' start={4.31} end={9.48} isDebugMode={true} muted={false} />)
        expect(getAllByText('main video')).toBeDefined()
    })
})
