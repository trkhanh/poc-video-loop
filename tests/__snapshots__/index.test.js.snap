// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoLooper matches Snapshot 1`] = `
<div>
  <div
    autoplay=""
    class="sc-bdVaJa hatrPw"
    end="9.48"
    height="100vh"
    speed="1"
    start="4.31"
    width="100%"
  >
    <div
      autoplay=""
      class="sc-htpNat jfHBEU"
      end="9.48"
      height="100vh"
      speed="1"
      start="4.31"
      width="100%"
    />
    <video
      autoplay=""
      class="sc-bwzfXH cMFZDI"
      end="9.48"
      height="100vh"
      speed="1"
      start="4.31"
      width="100%"
    >
      <source
        src="sample.mp4"
        type="video/mp4"
      />
    </video>
    <video
      autoplay=""
      class="sc-bwzfXH eqeTFC videoClone"
      end="9.48"
      height="100vh"
      speed="1"
      start="4.31"
      width="100%"
    >
      <source
        src="sample.mp4"
        type="video/mp4"
      />
    </video>
  </div>
</div>
`;